File: events-planner/frontend/src/api/events.js
export async function fetchEvents({page=1, per=10, q=''}){
  const token = localStorage.token
  const res = await fetch(`/api/events?page=${page}&per_page=${per}&q=${q}`, {
    headers:{ Authorization:'Bearer '+token }
  })
  return await res.json()
}

export async function createEvent(data){
  const token = localStorage.token
  const res = await fetch('/api/events',{
    method:'POST',
    headers:{
      'Content-Type':'application/json',
      Authorization:'Bearer '+token
    },
    body:JSON.stringify(data)
  })
  return await res.json()
}

export async function updateEvent(id, data){
  const token = localStorage.token
  const res = await fetch(`/api/events/${id}`,{
    method:'PUT',
    headers:{
      'Content-Type':'application/json',
      Authorization:'Bearer '+token
    },
    body:JSON.stringify(data)
  })
  return await res.json()
}

export async function deleteEvent(id){
  const token = localStorage.token
  await fetch(`/api/events/${id}`,{
    method:'DELETE',
    headers:{ Authorization:'Bearer '+token }
  })
}

export async function fetchSharedEvent(uuid){
  const res = await fetch(`/api/events/share/${uuid}`)
  return await res.json()
}

File: events-planner/backend/routes/events.py
# backend/routes/events.py
from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity
from models import Event, Task, db
from datetime import date

bp = Blueprint('events', __name__, url_prefix='/api/events')

@bp.route('', methods=['GET'])
@jwt_required()
def list_events():
    uid = get_jwt_identity()
    q = Event.query.filter(
        (Event.owner_id == uid) | (Event.date < date.today())
    )
    term = request.args.get('q', '').strip()
    if term:
        q = q.filter(Event.title.ilike(f'%{term}%'))
    page = int(request.args.get('page', 1))
    per = int(request.args.get('per_page', 10))
    pag = q.paginate(page=page, per_page=per, error_out=False)
    return jsonify({
        "items": [e.serialize() for e in pag.items],
        "total": pag.total,
        "page": pag.page
    })

@bp.route('', methods=['POST'])
@jwt_required()
def create_event():
    data = request.json
    uid = get_jwt_identity()
    ev = Event(**data, owner_id=uid)
    db.session.add(ev)
    db.session.commit()
    return jsonify(ev.serialize()), 201

@bp.route('/<int:id>', methods=['PUT', 'DELETE'])
@jwt_required()
def mod_event(id):
    uid = get_jwt_identity()
    ev = Event.query.get_or_404(id)
    if ev.owner_id != uid:
        return '', 403
    if request.method == 'PUT':
        for k, v in request.json.items():
            setattr(ev, k, v)
        db.session.commit()
        return jsonify(ev.serialize())
    db.session.delete(ev)
    db.session.commit()
    return '', 204

@bp.route('/share/<share_uuid>', methods=['GET'])
def public_event(share_uuid):
    ev = Event.query.filter_by(share_uuid=share_uuid).first_or_404()
    tasks = [t.serialize() for t in Task.query.filter_by(event_id=ev.id)]
    out = ev.serialize()
    out['tasks'] = tasks
    return jsonify(out)
